src/qtt/algorithms/random_telegraph_signal.py

```diff
@@ -243,9 +243,10 @@ def tunnelrates_RTS(data: Union[np.ndarray, qcodes.data.data_set.DataSet], sampl
 
     # plotting a 2d histogram of the RTS
     if fig:
-        xdata = np.array(range(0, len(data))) / samplerate * 1000
-        Z, xedges, yedges = np.histogram2d(xdata, data, bins=[int(
-            np.sqrt(len(xdata)) / 2), int(np.sqrt(len(data)) / 2)])
+        xdata = np.arange(len(data)) / (samplerate / 1000)
+        ny = min(int(np.sqrt(len(data))/2), 800)
+        nx = min(int(np.sqrt(len(data))/2), 1200)
```

Comment:
data should be xdata

Reply:
xdata and data have the same size, so it is ok. Will change it anyway